src/main/java/com/epam/esm/controller/TagsController.java

```diff
@@ -1,66 +1,70 @@
-     * @return ResponseEntity<?> A response entity representing the result of the creation operation.
-     *      It could contain the created tag, a custom message, or nothing (in case of void).
-     *      if tag name is not valid, returns bad request
-     *      if tag already exists, returns bad request
-     *      if tag does not exist, but cannot be saved, return bad request
-     *      if it is saved, return CREATED and tag saved
-     * @PostMapping This annotation maps HTTP POST requests onto this method.
-     * @value "/tag" The path where this method is mapped.
-     * @consumes {"application/json"} Specifies that this method only processes requests where the Content-Type header is application/json.
-     * @produces {"application/json"} Specifies that this method returns data in application/json format.
-     */
-    @PostMapping(value = "/tag", consumes = {"application/json"}, produces = {"application/json"})
-    ResponseEntity<?> postTag(@RequestBody TagRequestDTO requestDTO) {
-        return tagService.saveTag(requestDTO.name());
+    @PostMapping(consumes = {"application/json"}, produces = {"application/json"})
+    public ResponseEntity<EntityModel<TagResponseDTO>> postTag(@RequestBody TagRequestDTO requestDTO) {
+        TagResponseDTO tagDTO = tagService.saveTag(requestDTO.name());
+
+        EntityModel<TagResponseDTO> resource = EntityModel.of(tagDTO);
+        resource.add(linkTo(TagsController.class).slash(tagDTO.id()).withSelfRel());
+
+        return ResponseEntity.status(CREATED).body(resource);
     }
 
-    /**
-     * Retrieves a tag by its ID.
-     *
-     * @param id The unique identifier of the tag to be retrieved.
-     * @return ResponseEntity<?> A response entity containing the tag with the given ID.
-     * can return the requested tag, or not found
-     * @GetMapping This annotation maps HTTP GET requests onto this method.
-     * @value "/tag/{id}" The path where this method is mapped. It includes a path variable 'id'.
-     * @consumes {"application/json"} Specifies that this method only processes requests where the Content-Type header is application/json.
-     * @produces {"application/json"} Specifies that this method returns data in application/json format.
-     */
-    @GetMapping(value = "/tag/{id}", consumes = {"application/json"}, produces = {"application/json"})
-    ResponseEntity<?> getTagById(@PathVariable long id) {
-        return tagService.getTag(id);
+    //Links to next page
+    @GetMapping
+    public ResponseEntity<PagedModel<EntityModel<TagResponseDTO>>> getAllTags(
+        @RequestParam(defaultValue = "0") int page,
+        @RequestParam(defaultValue = "10") int size,
+        @RequestParam(defaultValue = "id") String sort) {
+
+        Pageable pageable = PageRequest.of(page, size, Sort.by(sort));
+        Page<TagResponseDTO> tagDTOPage = tagService.getAllTags(pageable);
+
+        PagedModel.PageMetadata pageMetadata = new PagedModel.PageMetadata(
```

Comment:
You are using here different approach for paging than in UserController or Certificates controller? Ideally they should be all uniform. (They use `PagedResourcesAssembler`, but you don't use it here)

Reply:
changed